paderbox/visualization/plot.py

```diff
@@ -846,3 +853,197 @@ def seq2seq_alignment(alignment, targets=None, decode=None, ax=None,
+        ax:
+        scale:
+        **kwargs:
+
+    """
+    import paderbox as pb
+    from paderbox.array.interval import ArrayInterval
+
+    def _convert_array_interval(array_interval):
+        """Converts an ArrayInterval into values for `left` and `width` that
+        can be passed to plt.hbar"""
+        left = [
+            interval[0] * scale
+            for interval in array_interval.normalized_intervals
+        ]
+        width = [
+            (interval[1] - interval[0]) * scale
+            for interval in array_interval.normalized_intervals
+        ]
+        return left, width
+
+    if isinstance(width, ArrayInterval):
+        assert left is None, left
+        left, width = _convert_array_interval(width)
+    elif isinstance(width, list) and isinstance(
+            width[0], ArrayInterval
+    ):
+        if not pb.utils.nested.nested_all(width, lambda x: isinstance(x, ArrayInterval)):
+            raise ValueError()
+        assert left is None, left
+        if not isinstance(y, Iterable):
+            y = [y] * len(width)
+        assert len(y) == len(width), (y, len(width))
+        left = []
+        new_width = []
+        new_y = []
+        for w, y_ in zip(width, y):
+            left_, width_ = _convert_array_interval(w)
+            left.extend(left_)
+            new_width.extend(width_)
+            new_y.extend([y_] * len(width_))
+        width = new_width
+        y = new_y
+    elif not isinstance(width, Iterable):
+        width = [width]
+    ax.barh(y, width, height, left, **kwargs)
+    return ax
+
+
+@create_subplot
+def axvspan(start, stop=None, ax=None, **kwargs):
+    """
+    Wrapper around `plt.axvspan` that supports
+    `pb.array.interval.ArrayInterval`s.
+
+    `plt.axvspan` doesn't support arrays as inputs, so we need to do a for-loop
+    over the intervals.
+
+    Args:
+        start: Scalar start value or an `ArrayInterval`
+        stop: Scalar stop value or `None` if `start` is an `ArrayInterval`.
+        ax:
+        **kwargs:
+
+    """
+    from paderbox.array.interval import ArrayInterval
+    if isinstance(start, ArrayInterval):
+        assert stop is None
+        for start, stop in start.intervals:
+            ax.axvspan(start, stop, **kwargs)
+    else:
+        ax.axvspan(start, stop, **kwargs)
+    return ax
+
+
+@create_subplot
+def activity(
+        activity_intervals,
+        y_labels=None,
+        segment_boundary_intervals=None,
+        ax=None,
+        activity_kwargs=None,
+        overlap_kwargs=None,
+        total_activity_kwargs=None,
+        plot_total_activity=True,
+        plot_overlaps=True,
+):
+    """
+    Plots an "activity plot" from a dictionary of
+    `pb.array.interval.ArrayInterval`s.
+
+    Args:
+        activity_intervals:
+        segment_boundary_intervals:
+        ax:
+        activity_kwargs:
+        overlap_kwargs:
+        total_activity_kwargs:
+        scale:
+        bar_edgecolor:
+        plot_total_activity:
+
```

Comment:
I think you need to add some explanation here, at least for activity_intervals and segment_boundary_intervals

Reply:
y_labels is missing